### PR TITLE
Add CURLStringFile class

### PR DESCRIPTION
**Pull request by ishanvyas22**
Fixes https://github.com/vimeo/psalm/issues/6414

### CODE REVIEW

File: dictionaries/CallMap_81_delta.php [hunk 1]
```diff
@@ -23,6 +23,7 @@
     'imagecreatefromavif' => ['false|GdImage', 'filename'=>'string'],
     'mysqli_fetch_column' => ['null|int|float|string|false', 'result'=>'mysqli_result', 'column='=>'int'],
     'mysqli_result::fetch_column' => ['null|int|float|string|false', 'column='=>'int'],
+    'CURLStringFile::__construct' => ['void', 'data'=>'string', 'postname'=>'string', 'mime'=>'string'],
```
Comment (orklah) [inline dictionaries/CallMap_81_delta.php:26]: The CallMap syntax uses an "=" at the end of an argument name to convey that the argument is optional. You should add that on `mime`

File: stubs/Php81.phpstub [hunk 2]
```diff
@@ -6,6 +6,16 @@ namespace {
     }
 }
 
+namespace {
+    class CURLStringFile {
+        public string $data;
+        public string $postname;
+        public string $mime;
+
+        public function __construct(string $data, string $postname, string $mime = "application/octet-stream") {}
```
Comment (orklah) [inline stubs/Php81.phpstub:15]: unless you have specific annotations to add, this should not be needed. The __construct signature is redundant with the CallMap and the properties signatures should be detailed in PropertyMap.php

**Review by orklah — CHANGES_REQUESTED** [review]
Great start! Here's some comments. You'll see the CI failing too because there are changes to be made in CallMap.php too. You'll find more details here: https://psalm.dev/docs/contributing/editing_callmaps/

**Comment by ishanvyas22** [comment]
@orklah, made changes according to your suggestion. Do I need to add properties in `PropertyMap.php` file too?

It seems it will not be needed because I don't see any properties added for `CURLFile` class for example.

**Comment by orklah** [comment]
Properties may be resolved through reflection, best way to be sure would be to run Psalm on 8.1 and check, but no need to go this far. If anyone has a need for those and they're not typed, they'll create an issue.

Thanks for the PR! If you're participating in Hacktoberfest, could you confirm this PR is counted? Otherwise I'll add the specific tag, I just want to check if this works without it.

**Comment by ishanvyas22** [comment]
Yes, it would be great if you add the label :). Otherwise it won't be counted.

**Pull request merged**